src/prerequisites/tests/test_signals.py

```diff
@@ -154,3 +156,36 @@ def test_update_cache_triggered_by_parent_object_deletion(self, task):
 
         quest.delete()  # doesn't call task because parent_object no longer exists.
         self.assertEqual(task.call_count, 1)
+
+    def test_on_quest_badge_save_with_rank_prereq__creation(self):
+        """ test that creating a prereq with rank requirements makes a new map """
+
+        # create prereq with rank as the first prereq
+        # and check if new map is created
+        first_rank = baker.make('courses.rank')
+        baker.make(
+            Prereq,
+            parent_object=baker.make('quest_manager.quest'),
+            prereq_object=first_rank
+        )
+
+        self.assertTrue(CytoScape.objects.filter(
+            name='New Map',
+            initial_content_type=ContentType.objects.get_for_model(first_rank),
+            initial_object_id=first_rank.id,
+        ).exists())
+
+        # create prereq with rank as the second prereq
+        # and check if new map is created
+        second_rank = baker.make('courses.rank')
+        baker.make(
+            Prereq,
+            parent_object=baker.make('quest_manager.quest'),
+            prereq_object=second_rank
+        )
+
+        self.assertTrue(CytoScape.objects.filter(
+            name='New Map',
+            initial_content_type=ContentType.objects.get_for_model(second_rank),
+            initial_object_id=second_rank.id,
+        ).exists())
```

Comment:
@coderabbitai we are not currently using type annotations in this codebase.  We prefer to use Django's built-in testing assertion methods instead of raw `assert`.